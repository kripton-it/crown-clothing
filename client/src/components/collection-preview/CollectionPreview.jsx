import React from "react";
import { withRouter } from "react-router-dom";

import CollectionItemContainer from "../collection-item/CollectionItemContainer";

import {
  CollectionPreviewContainer,
  CollectionTitle,
  CollectionList
} from "./CollectionPreviewStyles";

const PREVIEWS_NUMBER = 4;

export const CollectionPreview = ({ collection, history, match, routName }) => {
  const { title, items } = collection;

  return (
    <CollectionPreviewContainer>
      <CollectionTitle
        onClick={() => history.push(`${match.path}/${routName}`)}
      >
        {title}
      </CollectionTitle>
      <CollectionList>
        {items.slice(0, PREVIEWS_NUMBER).map(item => (
          <CollectionItemContainer key={item.id} item={item} />
        ))}
      </CollectionList>
    </CollectionPreviewContainer>
  );
};

export default withRouter(CollectionPreview);
