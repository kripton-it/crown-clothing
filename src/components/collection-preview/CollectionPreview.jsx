import React from "react";

import CollectionItem from "../collection-item/CollectionItem";

import "./CollectionPreview.styles.scss";

const PREVIEWS_NUMBER = 4;

const CollectionPreview = ({ title, items }) => {
  return (
    <div className="collection-preview">
      <h1 className="title">{title}</h1>
      <div className="preview">
        {items.slice(0, PREVIEWS_NUMBER).map(item => (
          <CollectionItem key={item.id} item={item} />
        ))}
      </div>
    </div>
  );
};

export default CollectionPreview;
