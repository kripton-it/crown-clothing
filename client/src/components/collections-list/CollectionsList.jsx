import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import CollectionPreview from "../collection-preview/CollectionPreview";

import { CollectionsListContainer } from "./CollectionsListStyles";

import { selectCollectionsForPreview } from "../../redux/shop/shop-selectors";

export const CollectionsList = ({ collections }) => {
  return (
    <CollectionsListContainer>
      {collections.map(collection => (
        <CollectionPreview key={collection.id} collection={collection} />
      ))}
    </CollectionsListContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  collections: selectCollectionsForPreview
});

export default connect(mapStateToProps)(CollectionsList);
