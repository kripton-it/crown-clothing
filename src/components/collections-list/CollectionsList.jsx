import React from 'react';
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import CollectionPreview from "../collection-preview/CollectionPreview";

import './CollectionsList.styles.scss'

import { selectCollections } from "../../redux/shop/shop-selectors";

const CollectionsList = ({collections}) => {
  return (
    <div className="collections-list">
      {collections.map(({ id, ...restProps }) => (
        <CollectionPreview key={id} {...restProps} />
      ))}
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  collections: selectCollections,
});

export default connect(mapStateToProps)(CollectionsList);