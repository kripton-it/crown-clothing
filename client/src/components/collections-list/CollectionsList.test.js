import React from "react";
import { shallow } from "enzyme";

import { CollectionsList } from "./CollectionsList";
import CollectionPreview from "../collection-preview/CollectionPreview";

describe("CollectionsList component", () => {
  const mockProps = {
    collections: [
      {
        id: 1,
        title: "test",
        items: []
      }
    ]
  };

  const wrapper = shallow(<CollectionsList {...mockProps} />);

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render an equal number of CollectionPreview components as the collections prop", () => {
    expect(wrapper.find(CollectionPreview).length).toEqual(
      mockProps.collections.length
    );
  });
});
