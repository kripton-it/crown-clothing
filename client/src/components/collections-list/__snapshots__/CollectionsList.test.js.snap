// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsList component should render 1`] = `ShallowWrapper {}`;
